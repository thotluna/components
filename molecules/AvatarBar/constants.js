export const options = {
  sizes: ['sm', 'lg'],
}
