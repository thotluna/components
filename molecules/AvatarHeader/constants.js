export const options = {
  size: ['sm', 'md'],
}
