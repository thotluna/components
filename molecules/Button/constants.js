export const options = {
  colors: ['base', 'primary', 'secondary'],
}
