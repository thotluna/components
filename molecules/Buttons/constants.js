export const options = {
  size: ['full', 'short', 'mini'],
  color: ['primary', 'secondary', 'inverted'],
}
