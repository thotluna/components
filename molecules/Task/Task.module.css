.task {
  display: flex;
  width: 100%;
  height: 51px;
  align-items: center;
  justify-content: space-between;
  padding: 0 1rem;
  border: 2px solid var(--color-primary);
  border-radius: var(--border-radius-md);
}

.wrapper {
  display: flex;
  width: 100%;
  align-items: center;
  justify-content: flex-start;
  gap: 1rem;
}
