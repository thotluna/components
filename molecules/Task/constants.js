export const states = {
  ACTIVE: 'active',
  PENDING: 'pending',
  CULMINATED: 'culminated',
  DELETE: 'delete',
}
export const options = {
  states: [states.ACTIVE, states.CULMINATED, states.DELETE, states.PENDING],
}
