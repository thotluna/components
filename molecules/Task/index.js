export { default } from './Task'
export { options } from './constants'
export { default as styles } from './Task.module.css'
