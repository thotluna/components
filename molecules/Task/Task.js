import React from 'react'
import PropTypes from 'prop-types'
import { styles, options } from '.'
import { getClasses } from '../../helpers/styles'

import Icon from '../../atoms/Icon'
import Title from '../../atoms/Title'
import Button from '../Button'
import { statusTask } from './constants'

export const Task = ({ children, status, handelOnDelete, handelOnCheck }) => {
  const getStyles = getClasses(styles)({ status })

  return (
    <div className={getStyles('task', ['status'])}>
      <div
        style={{
          display: 'flex',
          alignItems: 'center',
          justifyContent: 'flex-start',
          gap: '2rem',
        }}
      >
        {(status === statusTask.FIRST ||
          status === statusTask.SECOND ||
          status === statusTask.PENDING) && <Icon type="drag" />}
        {status === statusTask.ACTIVE && (
          <Button
            id="Delete task"
            ariaLabel="Delete task"
            icon="circle"
            onClick={handelOnCheck}
            value=""
          />
        )}
        {status === statusTask.CULMINATED && <Icon type="check" />}
        <Title size="md">{children + ' ' + status}</Title>
      </div>
      {(status === statusTask.FIRST ||
        status === statusTask.SECOND ||
        status === statusTask.PENDING) && (
        <Button
          id="Delete task"
          ariaLabel="Delete task"
          icon="trash"
          onClick={handelOnDelete}
          value=""
        />
      )}
    </div>
  )
}

Task.propTypes = {
  children: PropTypes.node.isRequired,
  handelOnDelete: PropTypes.func.isRequired,
  handelOnCheck: PropTypes.func.isRequired,
  status: PropTypes.oneOf(options.status),
}

Task.defaultProps = {
  status: statusTask.ACTIVE,
}

export default Task
