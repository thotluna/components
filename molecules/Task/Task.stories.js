import React from 'react'
import Task from './Task'
import { options } from './constants'

import { getVariusTemplate, getOptionsArgTypes } from '../../helpers/storybook'

const Template = (args) => <Task {...args} />
const ListTemplate = getVariusTemplate(Task)

export default {
  title: 'Molecules/Task',
  component: Task,
  args: {
    id: 'default',
    children: 'Sir Goblin-town feels characters ruined precautions.',
    defaultChecked: false,
    onCheck: () => {},
    onDelete: () => {},
  },
  argTypes: {
    type: getOptionsArgTypes(options.types),
    onCheck: { defaultValue: null },
    onDelete: { defaultValue: null },
  },
}

export const Default = Template.bind({})

export const Long = Template.bind({})
Long.args = {
  id: 'long',
  children:
    "Swords are no more use here. Moments shiny rings Cirith nowhere garrison ax pointy endless mention trousers. Retribution Frodo's wrong? Them leaf question require spared houses sapphire Balin. Kind pouring LothlÃ³rien ragged retribution covers lair less goings recognizes unleash. Whence Precious token hidden hire bandy make farming. Shards farmer unfriendly handle Bain gongs. Mend recall sorry oncoming roll unmade cries visitors Lobelia Sackville-Baggins was bidden. Dank rather between short doomed Mellon tomb consequence girl crescent attacking.",
}

export const CheckedTasks = Template.bind({})
CheckedTasks.args = {
  defaultChecked: true,
  id: `checked-true`,
}

export const Pending = Template.bind({})
Pending.args = {
  id: `checked-Pending`,
  isPending: true,
}

export const Types = ListTemplate.bind({})
Types.args = {
  items: options.types.map((type) => ({
    type,
    id: `type-${type}`,
  })),
}

export const PendingTypes = ListTemplate.bind({})
PendingTypes.args = {
  items: options.types.map((type) => ({
    type,
    isPending: true,
    id: `pending-${type}`,
  })),
}
