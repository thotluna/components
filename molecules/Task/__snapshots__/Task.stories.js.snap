// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Molecules/Task Checked Tasks 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="container"
  >
    <div
      className="card color-base size-sm is-clickable"
      onClick={[Function]}
      tabIndex={0}
    >
      <div
        className="task"
      >
        <div
          className="content"
        >
          <div
            className="container-check size-sm"
          >
            <input
              aria-label="check: Sir Goblin-town feels characters ruined precautions."
              className="input-check color-primary size-sm"
              id="card-checked-true"
              tabIndex={0}
              type="checkbox"
            />
          </div>
          <div
            className="spacer vertical"
            style={
              Object {
                "height": "100%",
                "maxHeight": "auto",
                "width": 5,
              }
            }
          />
          <p
            className="container color-base size-md"
          >
            Sir Goblin-town feels characters ruined precautions.
          </p>
        </div>
        <div
          className="spacer vertical"
          style={
            Object {
              "height": "100%",
              "maxHeight": "auto",
              "width": 15,
            }
          }
        />
        <div
          onClick={[Function]}
        >
          <button
            aria-label="delete-task"
            className="buttons size-mini color-inverted is-only-icon"
            onClick={[Function]}
          >
            <div
              className="container color-primary size-sm"
            >
              <svg
                fill="none"
                height="100%"
                stroke="currentColor"
                viewBox="0 0 24 24"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M19 7l-.867 12.142A2 2 0 0116.138 21H7.862a2 2 0 01-1.995-1.858L5 7m5 4v6m4-6v6m1-10V4a1 1 0 00-1-1h-4a1 1 0 00-1 1v3M4 7h16"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth={1}
                />
              </svg>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots Molecules/Task Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="container"
  >
    <div
      className="card color-base size-sm is-clickable"
      onClick={[Function]}
      tabIndex={0}
    >
      <div
        className="task"
      >
        <div
          className="content"
        >
          <div
            className="container-check size-sm"
          >
            <input
              aria-label="check: Sir Goblin-town feels characters ruined precautions."
              className="input-check color-primary size-sm"
              id="card-default"
              tabIndex={0}
              type="checkbox"
            />
          </div>
          <div
            className="spacer vertical"
            style={
              Object {
                "height": "100%",
                "maxHeight": "auto",
                "width": 5,
              }
            }
          />
          <p
            className="container color-base size-md"
          >
            Sir Goblin-town feels characters ruined precautions.
          </p>
        </div>
        <div
          className="spacer vertical"
          style={
            Object {
              "height": "100%",
              "maxHeight": "auto",
              "width": 15,
            }
          }
        />
        <div
          onClick={[Function]}
        >
          <button
            aria-label="delete-task"
            className="buttons size-mini color-inverted is-only-icon"
            onClick={[Function]}
          >
            <div
              className="container color-primary size-sm"
            >
              <svg
                fill="none"
                height="100%"
                stroke="currentColor"
                viewBox="0 0 24 24"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M19 7l-.867 12.142A2 2 0 0116.138 21H7.862a2 2 0 01-1.995-1.858L5 7m5 4v6m4-6v6m1-10V4a1 1 0 00-1-1h-4a1 1 0 00-1 1v3M4 7h16"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth={1}
                />
              </svg>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots Molecules/Task Long 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="container"
  >
    <div
      className="card color-base size-sm is-clickable"
      onClick={[Function]}
      tabIndex={0}
    >
      <div
        className="task"
      >
        <div
          className="content"
        >
          <div
            className="container-check size-sm"
          >
            <input
              aria-label="check: Swords are no more use here. Moments shiny rings Cirith nowhere garrison ax pointy endless mention trousers. Retribution Frodo's wrong? Them leaf question require spared houses sapphire Balin. Kind pouring Lothlórien ragged retribution covers lair less goings recognizes unleash. Whence Precious token hidden hire bandy make farming. Shards farmer unfriendly handle Bain gongs. Mend recall sorry oncoming roll unmade cries visitors Lobelia Sackville-Baggins was bidden. Dank rather between short doomed Mellon tomb consequence girl crescent attacking."
              className="input-check color-primary size-sm"
              id="card-long"
              tabIndex={0}
              type="checkbox"
            />
          </div>
          <div
            className="spacer vertical"
            style={
              Object {
                "height": "100%",
                "maxHeight": "auto",
                "width": 5,
              }
            }
          />
          <p
            className="container color-base size-md"
          >
            Swords are no more use here. Moments shiny rings Cirith nowhere garrison ax pointy endless mention trousers. Retribution Frodo's wrong? Them leaf question require spared houses sapphire Balin. Kind pouring Lothlórien ragged retribution covers lair less goings recognizes unleash. Whence Precious token hidden hire bandy make farming. Shards farmer unfriendly handle Bain gongs. Mend recall sorry oncoming roll unmade cries visitors Lobelia Sackville-Baggins was bidden. Dank rather between short doomed Mellon tomb consequence girl crescent attacking.
          </p>
        </div>
        <div
          className="spacer vertical"
          style={
            Object {
              "height": "100%",
              "maxHeight": "auto",
              "width": 15,
            }
          }
        />
        <div
          onClick={[Function]}
        >
          <button
            aria-label="delete-task"
            className="buttons size-mini color-inverted is-only-icon"
            onClick={[Function]}
          >
            <div
              className="container color-primary size-sm"
            >
              <svg
                fill="none"
                height="100%"
                stroke="currentColor"
                viewBox="0 0 24 24"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M19 7l-.867 12.142A2 2 0 0116.138 21H7.862a2 2 0 01-1.995-1.858L5 7m5 4v6m4-6v6m1-10V4a1 1 0 00-1-1h-4a1 1 0 00-1 1v3M4 7h16"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth={1}
                />
              </svg>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots Molecules/Task Pending 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="container"
  >
    <div
      className="card color-base size-sm is-draggable"
      onClick={[Function]}
      tabIndex={-1}
    >
      <div
        className="task"
      >
        <div
          className="content"
        >
          <div
            className="container color-sun size-md"
          >
            <svg
              fill="none"
              height="80%"
              stroke="currentColor"
              viewBox="0 0 24 24"
              width="80%"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7 11.5V14m0-2.5v-6a1.5 1.5 0 113 0m-3 6a1.5 1.5 0 00-3 0v2a7.5 7.5 0 0015 0v-5a1.5 1.5 0 00-3 0m-6-3V11m0-5.5v-1a1.5 1.5 0 013 0v1m0 0V11m0-5.5a1.5 1.5 0 013 0v3m0 0V11"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth={1}
              />
            </svg>
          </div>
          <div
            className="spacer vertical"
            style={
              Object {
                "height": "100%",
                "maxHeight": "auto",
                "width": 5,
              }
            }
          />
          <p
            className="container color-base size-md"
          >
            Sir Goblin-town feels characters ruined precautions.
          </p>
        </div>
        <div
          className="spacer vertical"
          style={
            Object {
              "height": "100%",
              "maxHeight": "auto",
              "width": 15,
            }
          }
        />
        <div
          onClick={[Function]}
        >
          <button
            aria-label="delete-task"
            className="buttons size-mini color-inverted is-only-icon"
            onClick={[Function]}
          >
            <div
              className="container color-primary size-sm"
            >
              <svg
                fill="none"
                height="100%"
                stroke="currentColor"
                viewBox="0 0 24 24"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M19 7l-.867 12.142A2 2 0 0116.138 21H7.862a2 2 0 01-1.995-1.858L5 7m5 4v6m4-6v6m1-10V4a1 1 0 00-1-1h-4a1 1 0 00-1 1v3M4 7h16"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth={1}
                />
              </svg>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots Molecules/Task Pending Types 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="container"
  >
    <div
      className="card color-base size-sm is-draggable"
      onClick={[Function]}
      tabIndex={-1}
    >
      <div
        className="task type-active"
      >
        <div
          className="content"
        >
          <div
            className="container color-sun size-md"
          >
            <svg
              fill="none"
              height="80%"
              stroke="currentColor"
              viewBox="0 0 24 24"
              width="80%"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7 11.5V14m0-2.5v-6a1.5 1.5 0 113 0m-3 6a1.5 1.5 0 00-3 0v2a7.5 7.5 0 0015 0v-5a1.5 1.5 0 00-3 0m-6-3V11m0-5.5v-1a1.5 1.5 0 013 0v1m0 0V11m0-5.5a1.5 1.5 0 013 0v3m0 0V11"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth={1}
              />
            </svg>
          </div>
          <div
            className="spacer vertical"
            style={
              Object {
                "height": "100%",
                "maxHeight": "auto",
                "width": 5,
              }
            }
          />
          <p
            className="container color-base size-md"
          >
            Sir Goblin-town feels characters ruined precautions.
          </p>
        </div>
        <div
          className="spacer vertical"
          style={
            Object {
              "height": "100%",
              "maxHeight": "auto",
              "width": 15,
            }
          }
        />
        <div
          onClick={[Function]}
        >
          <button
            aria-label="delete-task"
            className="buttons size-mini color-inverted is-only-icon"
            onClick={[Function]}
          >
            <div
              className="container color-primary size-sm"
            >
              <svg
                fill="none"
                height="100%"
                stroke="currentColor"
                viewBox="0 0 24 24"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M19 7l-.867 12.142A2 2 0 0116.138 21H7.862a2 2 0 01-1.995-1.858L5 7m5 4v6m4-6v6m1-10V4a1 1 0 00-1-1h-4a1 1 0 00-1 1v3M4 7h16"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth={1}
                />
              </svg>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="card color-base size-sm is-draggable"
      onClick={[Function]}
      tabIndex={-1}
    >
      <div
        className="task type-standby"
      >
        <div
          className="content"
        >
          <div
            className="container color-sun size-md"
          >
            <svg
              fill="none"
              height="80%"
              stroke="currentColor"
              viewBox="0 0 24 24"
              width="80%"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7 11.5V14m0-2.5v-6a1.5 1.5 0 113 0m-3 6a1.5 1.5 0 00-3 0v2a7.5 7.5 0 0015 0v-5a1.5 1.5 0 00-3 0m-6-3V11m0-5.5v-1a1.5 1.5 0 013 0v1m0 0V11m0-5.5a1.5 1.5 0 013 0v3m0 0V11"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth={1}
              />
            </svg>
          </div>
          <div
            className="spacer vertical"
            style={
              Object {
                "height": "100%",
                "maxHeight": "auto",
                "width": 5,
              }
            }
          />
          <p
            className="container color-base size-md"
          >
            Sir Goblin-town feels characters ruined precautions.
          </p>
        </div>
        <div
          className="spacer vertical"
          style={
            Object {
              "height": "100%",
              "maxHeight": "auto",
              "width": 15,
            }
          }
        />
        <div
          onClick={[Function]}
        >
          <button
            aria-label="delete-task"
            className="buttons size-mini color-inverted is-only-icon"
            onClick={[Function]}
          >
            <div
              className="container color-primary size-sm"
            >
              <svg
                fill="none"
                height="100%"
                stroke="currentColor"
                viewBox="0 0 24 24"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M19 7l-.867 12.142A2 2 0 0116.138 21H7.862a2 2 0 01-1.995-1.858L5 7m5 4v6m4-6v6m1-10V4a1 1 0 00-1-1h-4a1 1 0 00-1 1v3M4 7h16"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth={1}
                />
              </svg>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots Molecules/Task Types 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="container"
  >
    <div
      className="card color-base size-sm is-clickable"
      onClick={[Function]}
      tabIndex={0}
    >
      <div
        className="task type-active"
      >
        <div
          className="content"
        >
          <div
            className="container-check size-sm"
          >
            <input
              aria-label="check: Sir Goblin-town feels characters ruined precautions."
              className="input-check color-primary size-sm"
              id="card-type-active"
              tabIndex={0}
              type="checkbox"
            />
          </div>
          <div
            className="spacer vertical"
            style={
              Object {
                "height": "100%",
                "maxHeight": "auto",
                "width": 5,
              }
            }
          />
          <p
            className="container color-base size-md"
          >
            Sir Goblin-town feels characters ruined precautions.
          </p>
        </div>
        <div
          className="spacer vertical"
          style={
            Object {
              "height": "100%",
              "maxHeight": "auto",
              "width": 15,
            }
          }
        />
        <div
          onClick={[Function]}
        >
          <button
            aria-label="delete-task"
            className="buttons size-mini color-inverted is-only-icon"
            onClick={[Function]}
          >
            <div
              className="container color-primary size-sm"
            >
              <svg
                fill="none"
                height="100%"
                stroke="currentColor"
                viewBox="0 0 24 24"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M19 7l-.867 12.142A2 2 0 0116.138 21H7.862a2 2 0 01-1.995-1.858L5 7m5 4v6m4-6v6m1-10V4a1 1 0 00-1-1h-4a1 1 0 00-1 1v3M4 7h16"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth={1}
                />
              </svg>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="card color-base size-sm is-clickable"
      onClick={[Function]}
      tabIndex={0}
    >
      <div
        className="task type-standby"
      >
        <div
          className="content"
        >
          <div
            className="container-check size-sm"
          >
            <input
              aria-label="check: Sir Goblin-town feels characters ruined precautions."
              className="input-check color-primary size-sm"
              id="card-type-standby"
              tabIndex={0}
              type="checkbox"
            />
          </div>
          <div
            className="spacer vertical"
            style={
              Object {
                "height": "100%",
                "maxHeight": "auto",
                "width": 5,
              }
            }
          />
          <p
            className="container color-base size-md"
          >
            Sir Goblin-town feels characters ruined precautions.
          </p>
        </div>
        <div
          className="spacer vertical"
          style={
            Object {
              "height": "100%",
              "maxHeight": "auto",
              "width": 15,
            }
          }
        />
        <div
          onClick={[Function]}
        >
          <button
            aria-label="delete-task"
            className="buttons size-mini color-inverted is-only-icon"
            onClick={[Function]}
          >
            <div
              className="container color-primary size-sm"
            >
              <svg
                fill="none"
                height="100%"
                stroke="currentColor"
                viewBox="0 0 24 24"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M19 7l-.867 12.142A2 2 0 0116.138 21H7.862a2 2 0 01-1.995-1.858L5 7m5 4v6m4-6v6m1-10V4a1 1 0 00-1-1h-4a1 1 0 00-1 1v3M4 7h16"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth={1}
                />
              </svg>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
